keras_cv/layers/preprocessing/togray.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+import tensorflow.keras.layers as layers
+from tensorflow.keras import layers, backend
+
+
+class ToGray(layers.Layer):
+    """ToGray class for transforming RGB image to Grayscale image. The expected images 
+    should be [0-255] pixel ranges.
+    Input shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Args:
+        seed:
+            Integer. Used to create a random seed.
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    to_grayscale = keras_cv.layers.preprocessing.ToGray()
+    augmented_images = to_grayscale(images)
+    ```
+    """
+
+    def __init__(self, num_output_channels=1, **kwargs):
+        super().__init__(**kwargs)
+        self.num_output_channels = num_output_channels
+
+    def _check_input_params(self, num_output_channels):
+        if num_output_channels not in [1, 3]:
+            raise ValueError(
+                f"Valid arugment for num_output_channels param are 1 or 3. Got {num_output_channels}"
+            )
+        self.num_output_channels = num_output_channels
+
+    def _rgb_to_grayscale(self, image):
+        if self.num_output_channels == 1:
+            return tf.image.rgb_to_grayscale(image)
+        elif self.num_output_channels == 3:
+            _grayscale = tf.image.rgb_to_grayscale(image)
+            return tf.concat([_grayscale, _grayscale, _grayscale], axis=-1)
```

Comment:
What's the use case for 3 channel grayscale?

Reply:
Let's say you want to train a model with RGB input and want to use this layer as an augmentation. Then you need to pass num_output_channels to 3. And similarly, for the case of grayscale input training, you will need to pass 1. FYI, `torch.transform` supports this way also.

Reply:
Good point, thanks.